### PR TITLE
Add type in PropertyKey

### PR DESCRIPTION
**Pull request by rongrong**
Add PropertyType in PropertyKey. Enforce property key type for all property keys, and validate values when they are set rather than retrieved. Store values in expected format so get is cheap to perform.

### CODE REVIEW

File: core/server/common/src/main/java/alluxio/conf/ServerConfiguration.java [hunk 1]
```diff
@@ -150,7 +149,7 @@ public static String get(PropertyKey key, ConfigurationValueOptions options) {
    * @param defaultValue the value to return if no value is set for the specified key
    * @return the value
    */
-  public static String getOrDefault(PropertyKey key, String defaultValue) {
+  public static Object getOrDefault(PropertyKey key, String defaultValue) {
```
Comment (beinan) [inline core/server/common/src/main/java/alluxio/conf/ServerConfiguration.java:152]: Any chance to make this one as public static <T> T getOrDefault(PropertyKey key, T defaultValue)?
Not sure if it's feasible or not....

File: core/common/src/main/java/alluxio/conf/InstancedConfiguration.java [hunk 2]
```diff
@@ -242,97 +249,47 @@ public void merge(Map<?, ?> properties, Source source) {
...
   @Override
   public long getBytes(PropertyKey key) {
-    String rawValue = get(key);
-
-    try {
-      return FormatUtils.parseSpaceSize(rawValue);
-    } catch (Exception ex) {
-      throw new RuntimeException(ExceptionMessage.KEY_NOT_BYTES.getMessage(rawValue, key));
-    }
+    return (Long) get(key);
```
Comment (beinan) [inline core/common/src/main/java/alluxio/conf/InstancedConfiguration.java:287]: So we're skipping the FormatUtils.parseSpaceSize,  will it work for something like '1GB'?

File: core/common/src/main/java/alluxio/conf/InstancedConfiguration.java [hunk 3]
```diff
@@ -242,97 +249,47 @@ public void merge(Map<?, ?> properties, Source source) {
...
 
   @Override
   public long getMs(PropertyKey key) {
-    String rawValue = get(key);
-    try {
-      return FormatUtils.parseTimeSize(rawValue);
-    } catch (Exception e) {
-      throw new RuntimeException(ExceptionMessage.KEY_NOT_MS.getMessage(rawValue, key));
-    }
+    return (Long) get(key);
```
Comment (beinan) [inline core/common/src/main/java/alluxio/conf/InstancedConfiguration.java:292]: will this work for the time duration such as 1min 1h and etc..

Comment (rongrong) [reply core/common/src/main/java/alluxio/conf/InstancedConfiguration.java:292]: same as data size, parsing is done at set. All parsing are done at set except CLASS and LIST type.

**Review by beinan — APPROVED** [review]
lgtm

**Review by apc999 — APPROVED** [review]

**Pull request merged**